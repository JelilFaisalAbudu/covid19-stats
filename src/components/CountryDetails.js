import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import '../css/CountryDetails.css';

const CountryDetails = ({ country }) => (
  <div className="table-wrapper">
    <div className="table-title">
      <h3>
        Covid-19 Data -
        {' '}
        {country.Country}
      </h3>
      <span>
        Last Update:
        {' '}
        {`${new Date(country.Date)}`}
      </span>
    </div>
    <table className="table-fill">
      <thead>
        <tr>
          <th className="text-left">Item</th>
          <th className="text-left">Figures</th>
        </tr>
      </thead>
      <tbody className="table-hover">
        <tr>
          <td className="text-left">New Confirmed</td>
          <td className="text-left">{country.NewConfirmed}</td>
        </tr>
        <tr>
          <td className="text-left">Total Confirmed</td>
          <td className="text-left">{country.TotalConfirmed}</td>
        </tr>
        <tr>
          <td className="text-left">New Deaths</td>
          <td className="text-left">{country.NewDeaths}</td>
        </tr>
        <tr>
          <td className="text-left">Total Deaths</td>
          <td className="text-left">{country.TotalDeaths}</td>
        </tr>
        <tr>
          <td className="text-left">New Recovered</td>
          <td className="text-left">{country.NewRecovered}</td>
        </tr>
        <tr>
          <td className="text-left">Total Recovered</td>
          <td className="text-left">{country.TotalRecovered}</td>
        </tr>
      </tbody>
    </table>
    <Link to="/">
      <button type="button" className="back-button">Back</button>
    </Link>
  </div>
);

const mapStateToPropsCountryDetails = (state, ownProps) => {
  const countryName = ownProps.match.params.countryCountry;
  return {
    country: state.dataState.data.Countries.find(country => country.Country === countryName),
  };
};

CountryDetails.propTypes = {
  country: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default connect(mapStateToPropsCountryDetails)(CountryDetails);
