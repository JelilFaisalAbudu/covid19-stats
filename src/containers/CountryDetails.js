import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import '../css/CountryDetails.css';
import doFetchData from '../redux/actions/dataActions';

const CountryDetails = ({ country }) => (
  <div className="table-wrapper">
    <div className="table-title">
      <h3>
        Covid-19 Data -
        {' '}
        {country.Country}
      </h3>
      <span>
        Last Update:
        {' '}
        {`${new Date(country.Date)}`}
      </span>
    </div>
    <table className="table-fill">
      <thead>
        <tr>
          <th className="text-left">Item</th>
          <th className="text-left">Figures</th>
        </tr>
      </thead>
      <tbody className="table-hover">
        <tr>
          <td className="text-left">New Confirmed</td>
          <td className="text-left">{country.NewConfirmed}</td>
        </tr>
        <tr>
          <td className="text-left">Total Confirmed</td>
          <td className="text-left">{country.TotalConfirmed}</td>
        </tr>
        <tr>
          <td className="text-left">New Deaths</td>
          <td className="text-left">{country.NewDeaths}</td>
        </tr>
        <tr>
          <td className="text-left">Total Deaths</td>
          <td className="text-left">{country.TotalDeaths}</td>
        </tr>
        <tr>
          <td className="text-left">New Recovered</td>
          <td className="text-left">{country.NewRecovered}</td>
        </tr>
        <tr>
          <td className="text-left">Total Recovered</td>
          <td className="text-left">{country.TotalRecovered}</td>
        </tr>
      </tbody>
    </table>
    <Link to="/">
      <button type="button" className="back-button">Back</button>
    </Link>
  </div>
);

const mapStateToPropsCountryDetails = (state, ownProps) => {
  const countryName = ownProps.match.params.countryCountry;
  const countries = state.dataState.data.Countries;

  return {
    country: countries.find(country => country.Country === countryName),
  };
};

const mapDispatchToPropsCountryList = dispatch => ({
  fetchData: () => dispatch(doFetchData()),
});

CountryDetails.propTypes = {
  country: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default connect(
  mapStateToPropsCountryDetails,
  mapDispatchToPropsCountryList,
)(CountryDetails);
