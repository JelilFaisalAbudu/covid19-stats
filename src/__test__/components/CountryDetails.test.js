import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import CountryDetails from '../../components/CountryDetails';


const country = {
  Country: 'ALA Aland Islands',
  CountryCode: 'AX',
  Slug: 'ala-aland-islands',
  NewConfirmed: 0,
  TotalConfirmed: 0,
  NewDeaths: 0,
  TotalDeaths: 0,
  NewRecovered: 0,
  TotalRecovered: 0,
  Date: '2020-04-05T06:37:00Z',
};

describe('Country Details', () => {
  test('renders the back button', () => {
    render(<CountryDetails country={country} />);
    const textNode = screen.getByText(/Back/i);
    expect(textNode).toBeInTheDocument();
  });

  test('renders the app component correctly', () => {
    const app = renderer.create(<CountryDetails country={country} />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
