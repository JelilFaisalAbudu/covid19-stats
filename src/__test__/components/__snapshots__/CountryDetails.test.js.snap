// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Details renders the app component correctly 1`] = `
<div
  className="table-wrapper"
>
  <div
    className="table-title"
  >
    <h3>
      Covid-19 Data -
       
      ALA Aland Islands
    </h3>
    <span>
      Last Update:
       
      Sun Apr 05 2020 06:37:00 GMT+0000 (Greenwich Mean Time)
    </span>
  </div>
  <table
    className="table-fill"
  >
    <thead>
      <tr>
        <th
          className="text-left"
        >
          Item
        </th>
        <th
          className="text-left"
        >
          Figures
        </th>
      </tr>
    </thead>
    <tbody
      className="table-hover"
    >
      <tr>
        <td
          className="text-left"
        >
          New Confirmed
        </td>
        <td
          className="text-left"
        >
          0
        </td>
      </tr>
      <tr>
        <td
          className="text-left"
        >
          Total Confirmed
        </td>
        <td
          className="text-left"
        >
          0
        </td>
      </tr>
      <tr>
        <td
          className="text-left"
        >
          New Deaths
        </td>
        <td
          className="text-left"
        >
          0
        </td>
      </tr>
      <tr>
        <td
          className="text-left"
        >
          Total Deaths
        </td>
        <td
          className="text-left"
        >
          0
        </td>
      </tr>
      <tr>
        <td
          className="text-left"
        >
          New Recovered
        </td>
        <td
          className="text-left"
        >
          0
        </td>
      </tr>
      <tr>
        <td
          className="text-left"
        >
          Total Recovered
        </td>
        <td
          className="text-left"
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      Back
    </button>
  </a>
</div>
`;
